### ISSUE TITLE
is image_weights automatically disabled in the last epochs?

### ISSUE THREAD

**Issue by tino926** [ultralytics/yolov5#11999]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I am new to using the `--image-weights` flag to train a model. The training process is going well except for the last 25 epochs. During these epochs, the training and validation losses continue to improve, but the mAPs start to decrease. I am wondering if there is some mechanism (such as `--image-weights`) that is automatically disabled in the last few epochs? (I have searched through the code but cannot find any related code)

![image](https://github.com/ultralytics/yolov5/assets/2639399/154ca94e-a8ad-41bc-9f1c-b12353bbd78e)
![image](https://github.com/ultralytics/yolov5/assets/2639399/68c041c9-7342-404a-94f3-ea2f54affc93)
![image](https://github.com/ultralytics/yolov5/assets/2639399/8007089d-d06d-40c1-96a9-8836906bc77d)

### Additional

the training command:
```bash
CUDA_VISIBLE_DEVICES=0 python train.py \
  --project ... --cfg ... --data ... --img 480 --batch-size 64 --workers 16 --weights ... \
  --hyp ./data/hyps/hyp.scratch-low_mosYes_0.0072.yaml \
  --image-weights \
  --patience 200 --epoch 400 \
  --name test_imWgt \
  --resume ".../last.pt"
```

the content in hyp.scratch-low_mosYes_0.0072.yaml
```yaml
lr0: 0.0072  # initial learning rate (SGD=1E-2, Adam=1E-3)
lrf: 0.01  # final OneCycleLR learning rate (lr0 * lrf)
momentum: 0.937  # SGD momentum/Adam beta1
weight_decay: 0.0005  # optimizer weight decay 5e-4
warmup_epochs: 0.0  # warmup epochs (fractions ok)
warmup_momentum: 0.8  # warmup initial momentum
warmup_bias_lr: 0.001  # warmup initial bias lr
box: 0.05  # box loss gain
cls: 0.5  # cls loss gain
cls_pw: 1.0  # cls BCELoss positive_weight
obj: 1.0  # obj loss gain (scale with pixels)
obj_pw: 1.0  # obj BCELoss positive_weight
iou_t: 0.20  # IoU training threshold
anchor_t: 4.0  # anchor-multiple threshold
# anchors: 3  # anchors per output layer (0 to ignore)
fl_gamma: 0.0  # focal loss gamma (efficientDet default gamma=1.5)
hsv_h: 0.015  # image HSV-Hue augmentation (fraction)
hsv_s: 0.7  # image HSV-Saturation augmentation (fraction)
hsv_v: 0.4  # image HSV-Value augmentation (fraction)
degrees: 0.0  # image rotation (+/- deg)
translate: 0.1  # image translation (+/- fraction)
scale: 0.5  # image scale (+/- gain)
shear: 0.0  # image shear (+/- deg)
perspective: 0.0  # image perspective (+/- fraction), range 0-0.001
flipud: 0.0  # image flip up-down (probability)
fliplr: 0.5  # image flip left-right (probability)
mosaic: 1.0  # image mosaic (probability)
mixup: 0.0  # image mixup (probability)
copy_paste: 0.0  # segment copy-paste (probability)
```

**Comment by glenn-jocher** [comment]
@tino926 the `--image-weights` flag is not automatically disabled in the last epochs. It continues to be applied during the entire training process. It's likely that other factors, such as learning rate, data augmentation, or model capacity, may be affecting your mAP in the end stages. Feel free to tweak these hyperparameters to further diagnose the issue. If you have other questions, check out the [Ultralytics Docs](https://docs.ultralytics.com/yolov5).
